### ISSUE TITLE
Switching between workspaces causes the focussed tab to be changed

### ISSUE THREAD

**Issue by vraravam** [zen-browser/desktop#3063]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

I setup 3 workspaces in my profile. Each workspace also has some different tabs as pinned tabs. I don't use Essential tabs since I want each workspace to be "sandboxed" and not share any session info across.

With the above setup, when I am on the 2nd pinned tab in workspace1, and I switch to workspace2 and back again, the tab that has focus in workspace1 is the 1st **unpinned** tab. I would have expected that the pinned tab should have retained focus in workspace1 (when I switch to wotkspace1 from another workspace)

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.19 (Firefox 132.0.1) (aarch64)

### What platform are you seeing the problem on?

macOS - aarch64

**Comment by nijin26** [comment]
@vraravam , I am also facing the same issue in Zen browser. This should not be the expected behavior while switching workspace.

Platform:
Ubuntu 22.04 LTS

**Comment by oxidizingkilometerwildcard** [comment]
I also have the same issue on windows 11.

**Comment by vraravam** [comment]
This seems to have been fixed in the latest twilight build (t5). But, when starting Zen for the first time after quitting, it doesn't seem to remember the tab that had the last focus for each workspace. Thus, it always opens the first unpinned tab (and if no unpinned was present, it opens a new tab).

**Comment by vraravam** [comment]
yes, this is still relevant in the latest twilight build: 1.7.2t (Firefox 134.0.2) (2025-01-22) (aarch64)

**Comment by Feinzer** [comment]
Running Beta (1.7.6b). This is still relevant. It happens even without per-workspace pinned tabs too.

Zen can remember the last tab when I do:

1. Having 2 tabs on workspace 1
2. Switching to workspace 2 manually
3. Going back to workspace 1 Zen remembers the tab I was on before

But I can reproduce the issue consistently by doing:

1. Having 2 tabs on workspace 1
2. Click a link that changes my workspace automatically with the option "Switch to workspace where container is set as default when opening container tabs" enabled
3. Going back to workspace 1 Zen does not remember and throws me to the first tab at the top of the list regardless of where I was before

**Comment by Feinzer** [comment]
Ok I believe I know what's happening.

If I have two workspaces and open a tab that will switch to workspace 2, said tab actually opens for a second on workspace 1, closes itself and opens on the appropriate workspace 2. Causing Zen to remember the tab that got closed before switching to workspace 2.

**Comment by vraravam** [comment]
yes - this issue is still persistent as of the latest twilight build (1.9.1t (Firefox 136.0.1) (2025-03-11) (aarch64))

**Comment by mr-cheffy** [comment]
## Closing this issue as part of a repo cleanup

Due to technical-debt we had in the past, we are currently clearing out old, duplicate, stale, and non-actionable issues to improve issue tracking and make it easier to focus on fixing real bugs.

If this issue is still relevant, please open a new issue using our updated issue template, which will help us categorize and address it more efficiently.

Thanks for your understanding and for helping us improve the project! 🚀

This issue can be reopened at: https://github.com/zen-browser/desktop/issues/new?template=bug_report.yml&title=Switching%20between%20workspaces%20causes%20the%20focussed%20tab%20to%20be%20changed%0A
